src/structures/linked-list.js

```diff
@@ -0,0 +1,226 @@
+            return false;
+        }
+
+        //move head to next node
+        this.head = this.head.next;
+        this.size--;
+
+        return true;
+    }
+
+    removeFromEnd() {
+        //if list is empty
+        if (this.size === 0) {
+            return false;
+        }
+        let current = this.head;
+        let previous;
+
+        //iterate to the end of the list
+        while (current.next) {
+            previous = current;
+            current = current.next;
+        }
+
+        //remove node
+        previous.next = null;
+        this.size--;
+
+        return true;
+    }
+
+    removeAt(index) {
+        //if index is out of range
+        if (index < 0 || index >= this.size) {
+            return false;
+        }
+
+        //if index is first position
+        if (index === 0) {
+            this.removeFromBeginning();
+            return true;
+        }
+        let current = this.head;
+        let previous;
+        let count = 0;
+
+        //iterate to the position to remove
+        while (count < index) {
+            previous = current;
+            current = current.next;
+            count++;
+        }
+
+        //remove node
+        previous.next = current.next;
+        this.size--;
+
+        return true;
+    }
+
+    indexOfFirst(data) {
+        let current = this.head;
+        let count = 0;
+        while (current) {
+            if (current.data === data) {
+
+                //return when data is found at first time
+                return count;
+            }
+            current = current.next;
+            count++;
+        }
+        return -1;
+    }
+
+    indexOfLast(data) {
+        let current = this.head;
+        let count = 0;
+        let index = -1;
+        while (current) {
+            if (current.data === data) {
+                //store the index at each iteration
+                index = count;
+            }
+            current = current.next;
+            count++;
+        }
+        //return the last index
+        return index;
+    }
+
+    isEmpty() {
+        return this.size === 0;
+    }
+
+    getSize() {
+        return this.size;
+    }
+
+    printList() {
+        let current = this.head;
+        let list = '';
+        list+='[';
+        while (current) {
+            list += current.data;
+            if (current.next) {
+                list += ',';
+            }
+            current = current.next;
+        }
+        list+=']';
+        console.log(list);
+    }
+
+    reverse() {
+        //three pointer approach
+        let current = this.head;
+        let previous = null;
+        let next = null;
+        while (current) {
+            next = current.next;
+            current.next = previous;
+            previous = current;
+            current = next;
+        }
+        this.head = previous;
+    }  
+
+}
+
+
+//sample code to test the linked list
+let list = new LinkedList();
+list.insertAtBeginning(1);
```

Comment:
comment the test cases
or
use a different file for test cases

Reply:
done.
can this PR still be merged after committing new change or I have to create a new one?